charmcraft/commands/build.py

```diff
@@ -55,13 +55,16 @@
 
 def _pip_needs_system():
     """Determine whether pip3 defaults to --user, needing --system to turn it off."""
-    try:
-        from pip.commands.install import InstallCommand
-
-        return InstallCommand().cmd_opts.get_option("--system") is not None
-    except (ImportError, AttributeError, TypeError):
-        # probably not the bionic pip version then
-        return False
+    cmd = [
+        "python3",
+        "-c",
+        (
+            "from pip.commands.install import InstallCommand; "
+            'assert InstallCommand().cmd_opts.get_option("--system") is not None'
+        ),
+    ]
+    proc = subprocess.run(cmd, capture_output=True)
```

Comment:
`capture_output` does not exist in Python 3.6. Why we need it anyway?

Reply:
For context, subprocess.run is processed using charmcraft's python interpreter (3.8), not the one on the system, the python in the arguments to subprocess.run is the one on the system

Reply:
Note that charmcraft may still be installed in Bionic from PyPI.

Reply:
I thought we agreed to baseline on 3.8, was that not the case? python3.8 is available on 18.04.

Reply:
Let's talk about this. It's not clear to me where/how we want charmcraft to run.